src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,279 @@
+import org.mockito.MockitoAnnotations;
+
+/** */
+public final class SearchServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private SearchServlet searchObject;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    searchObject = new SearchServlet();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Test
+  /* Ensure the correct filters are set when all search parameters are specified. */
+  public void makeFiltersAllParams() {
+
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("courseName")).thenReturn("CS 105");
+    when(request.getParameter("profName")).thenReturn("Smith");
+    when(request.getParameter("term")).thenReturn("Spring 2020");
+    when(request.getParameter("units")).thenReturn("1,2,3");
+
+    List<Filter> filters = searchObject.getFilters(request);
+
+    List<Filter> expectedFilters = new ArrayList<>();
+    List<Integer> units = new ArrayList<>();
+    units.add(1);
+    units.add(2);
+    units.add(3);
+    expectedFilters.add(new FilterPredicate("name", FilterOperator.EQUAL, "CS 105"));
+    expectedFilters.add(new FilterPredicate("professor", FilterOperator.EQUAL, "Smith"));
+    expectedFilters.add(new FilterPredicate("term", FilterOperator.EQUAL, "Spring 2020"));
+    expectedFilters.add(new FilterPredicate("units", FilterOperator.IN, units));
+
+    assertEquals(filters, expectedFilters);
+  }
+
+  @Test
+  /* Ensure no filters are made given no search parameters are specified. */
+  public void makeFiltersNoParams() {
+
+    request = Mockito.mock(HttpServletRequest.class);
+
+    // TODO: Find a better way to replicate null parameters -
+    // getting NullPointerException when trying other methods.
+    when(request.getParameter("courseName")).thenReturn("");
+    when(request.getParameter("profName")).thenReturn("");
+    when(request.getParameter("term")).thenReturn("select");
+    when(request.getParameter("units")).thenReturn("");
+
+    List<Filter> filters = searchObject.getFilters(request);
+
+    List<Filter> expectedFilters = new ArrayList<>();
+
+    assertEquals(filters, expectedFilters);
+  }
+
+  @Test
+  /* Ensure the correct results are retrieved from Datastore given that all filters are set. */
+  public void getResultsAllFilters() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("courseName")).thenReturn("CS 105");
+    when(request.getParameter("profName")).thenReturn("Smith");
+    when(request.getParameter("term")).thenReturn("Spring 2020");
+    when(request.getParameter("units")).thenReturn("1,2,3");
+
+    List<Entity> expectedResults = new ArrayList<>();
+    Entity entity1 = new Entity("Course");
```

Comment:
probably make a function to create entities as well because it seems like you make a lot of them